### DISCUSSION TITLE
ESP8266 with 240x240 sprites

### DISCUSSION THREAD

**Discussion by error105** [Bodmer/TFT_eSPI#2156] [Q&A - General]
Hi, i want make gauges on my IPS GC9A01 with ESP8266 but propably the problem is with my ram. I want crate Sprites 240x240, how can i reduce my ColorDepth to show sprites with my 240x240 ?

`void createBackground() {
  gaugeBack.createSprite(190, 190);
  gaugeBack.setPivot(120, 120);
  tft.setPivot(160, 120);
  gaugeBack.fillSprite(TFT_TRANSPARENT);
  gaugeBack.pushImage(0, 0, 240, 240, SpeedoWHT);

  if (millis() >+ 5000){
  gaugeBack.setTextColor(COLORS);
  gaugeBack.setTextDatum(MC_DATUM);
  gaugeBack.setTextSize(3);
  gaugeBack.drawNumber(obroty, 120, 150, 1);
  }`

**Comment by Bodmer** [comment]
An ESP8266 can only spare about 40kbytes for sprties. If you are using the WiFi and buffering data the spare RAM will be smaller.

A 1 bit per pixel sprite would require 240*240/8 = 7200 bytes
A 4 bit per pixel sprite would require 240*240/2 = 28,800 bytes
A 8 bit per pixel sprite would require 249x240 = 57,600 bytes

This means at best you can use a 4 bit sprite. This means you then have to use a colour palette.

See examples for setting bit depth and managing the palette. [This example](https://github.com/Bodmer/TFT_eSPI/blob/master/examples/Sprite/Sprite_draw_4bit/Sprite_draw_4bit.ino) would be a good place to start, line 53 sets the colour depth.

**Comment by error105** [comment]
So, if im use Inkscape how i can export file in 4bit ?
And how convert png/jpg to 4 bit C in https://javl.github.io/image2cpp/ ?

Because when im change png to 4bit, and im set 4bit in program i dont see graphic :(

**Reply by Bodmer** [reply]
I am not familiar with inkscape so cannot help.

Kate Morris added the 4 bt sprite code and also created a python script that converts a BMP file to a 4 bit array. I have not used it so am unable to help but the python script is here:

https://github.com/Bodmer/TFT_eSPI/tree/master/Tools/bmp2array4bit

**Reply by error105** [reply]
After that i have just errorr when compiling :(
:
no matching function for call to 'TFT_eSprite::pushImage(int, int, int, int, const uint8_t [28800])'
